torchgeo/datasets/geo.py

```diff
@@ -722,7 +722,7 @@ def __getitem__(self, query: BoundingBox) -> dict[str, Any]:
         # Use array_to_tensor since rasterize may return uint16/uint32 arrays.
         masks = array_to_tensor(masks)
 
-        sample = {"mask": masks, "crs": self.crs, "bbox": query}
+        sample = {"mask": masks.long(), "crs": self.crs, "bbox": query}
```

Comment:
Why is this needed?

Reply:
Otherwise I got error where masks byte tensor is used in cross entropy loss which expects long tensor.

RasterDataset has a dtype function that forces it to be either long tensor or float32 tensor depending on whether the raster represents class labels or image.

Reply:
Can we add a similar dtype function to VectorDataset? By default, it will always be long (there shouldn't be any image vector datasets).